src/index.js

```diff
@@ -1 +1,46 @@
-export default "TODO";
+import * as babylon from "babylon";
+
+// Parse options values.
+const _parseOpts = (opts) => Object.keys(opts)
+  // Parse the matched replacement expression and
+  // https://github.com/babel/babylon/issues/210
+  .map((key) =>
+    [key, babylon.parse(`expr=${opts[key]}`).program.body.pop().expression.right])
+  // Objectify.
+  .reduce((memo, pair) => {
+    memo[pair[0]] = pair[1];
+    return memo;
+  }, {});
+
+// Parse, cache, and get options
+const _getOpts = function (ctxt, opts) {
+  ctxt._opts = ctxt._opts || _parseOpts(opts);
+  return ctxt._opts;
+};
+
+export default function ({ types: t }) {
+  return {
+    visitor: {
+      // require("TOKEN/lodash")
+      CallExpression(path, state) {
+        const node = path.node;
+        const args = node.arguments || [];
+        const opts = _getOpts(this, state.opts);
+
+        if (node.callee.name === "require" && args.length === 1 && t.isStringLiteral(args[0])) {
+          const src = args[0].value;
+          const [first, ...rest] = src.split("/");
+          const mod = (rest || []).join("/");
+
+          // Parse options and mutate with first match.
+          const replace = opts[Object.keys(opts).find((k) => k === first)];
+          if (replace) {
+            // Rewrite the `require`, then the module argument with stripped module path.
+            node.callee = replace;
```

Comment:
One other thought here - you probably want the value of `node.callee.loc` to remain, despite `node.callee` being replaced.  Otherwise you'll see some strange source-map action.  You could do something like:

```javascript
const loc = node.callee.loc;
node.callee = _.chain(replace).clone().assign({ loc }).value();
```

You'd still have invalid `loc` values in nested nodes, but so long as the `babylon.parse` was not provided a filename to attach to the nodes, the information would probably be discarded in a webpack-like build context.